### PR TITLE
fix(DatePicker): update range edges according to min and max boundaries

### PR DESCRIPTION
**Pull request by sun-tea**
- [ ] Feature
- [x] Fix
- [ ] Enhancement

## Description
Avoid this: 
Screenshot 2019-07-01 at 15 32 25
If the component is provided with an interval that is outside the minDate and maxDate, the displayed range should stay within these boundaries.

You can play with the stories to check if everything works as expected.

## Todo - Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.

### CODE REVIEW

File: src/DatePicker/__stories__/DatePicker.stories.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ import DatePickerReadme from '../README.md';
 
 const onChangeAction = action('onChange');
 const today = new Date();
+const debutOfToday = new Date(today.getFullYear(), today.getMonth(), today.getDate(), 0, 0, 0);
```
Comment (Artikodin) [inline src/DatePicker/__stories__/DatePicker.stories.js]: `debutOfToday` sound weird does it possible to change the naming with something like `todayBeginning` ?

**Comment by sun-tea** [comment]
@spartDev do you approve?

**Comment by sun-tea** [comment]
This component needs to be reworked to change its behaviour cf https://tillersystems.atlassian.net/browse/BP-508 https://tillersystems.atlassian.net/browse/BP-366

**Comment by sun-tea** [comment]
This component needs to be reworked to change its behaviour cf https://tillersystems.atlassian.net/browse/BP-508 https://tillersystems.atlassian.net/browse/BP-366

**Pull request closed without merging**